From 8a457c8d2a36b7173881de6610499a4060d67f60 Mon Sep 17 00:00:00 2001
From: Alastair D'Silva <alastair@d-silva.org>
Date: Thu, 17 Sep 2026 10:10:00 +1000
Subject: [PATCH 1/6] fix(allwinner): a733: disable SCPI in favor of native
 PSCI

The Allwinner A733 handles CPU and cluster power domain transitions
natively via the NCAT2 PPU and RTC watchdog in sunxi_native_pm_ncat2.
It does not run SCPI firmware on the companion ARISC coprocessor.

Explicitly set SUNXI_PSCI_USE_SCPI := 0 to avoid pulling in the unused
SCPI driver from allwinner-common.mk.

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>
---
 plat/allwinner/sun60i_a733/platform.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/allwinner/sun60i_a733/platform.mk b/plat/allwinner/sun60i_a733/platform.mk
index 0c8f6c0a06..a3010bf941 100644
--- a/plat/allwinner/sun60i_a733/platform.mk
+++ b/plat/allwinner/sun60i_a733/platform.mk
@@ -13,6 +13,8 @@ SUNXI_CPU_COUNT := 8
 
 SUNXI_SETUP_REGULATORS := 0
 
+# Native PSCI, no SCP
+SUNXI_PSCI_USE_SCPI     :=      0
 SUNXI_PSCI_USE_NATIVE   :=      1
 
 # The differences between the platforms are covered by the include files.
-- 
2.53.0

